geeservermap/layers.py

```diff
@@ -119,17 +161,11 @@ class MapLayer:
     def __init__(self, url, opacity, visible, attribution=ATTR):
         """TODO Missing docstring."""
         self.url = url
-        if opacity > 1:
-            print("opacity cannot be greater than 1, setting to 1")
-            opacity = 1
-        elif opacity < 0:
-            print("opacity cannot be less than 0, setting to 0")
-            opacity = 0
-        self.opacity = opacity
+        self.opacity = max(min(opacity, 1), 0)
```

Comment:
TBH I think it isn't worth the extra lines, this clamp mechanism is super simple to read and it will reduce the number of if statements to check in the coverage. Is it a problem if when a user sets opacity to 14, we silently agree it's dum and set it to 1? 
In your implementation the map is displayed anyway so apart from advanced users nobody will see the actual log.

Reply:
🤔 I don't think only advanced users will see it, since it's not going to the flask log but to the interpreter stdout (considering users read it, which is not always the case)... maybe we could use the `logging` module.  We could also reduce the number of `if` statements by checking both greater and lower at the same time, like

``` python
if opacity > 1 or opacity < 0:
    print("opacity cannot be greater than 1 or lower than 0, setting to 1")
    opacity = 1
```
I know is an extra statement just to send some message to the user.. I guess it's a matter of deciding where we consider the trade off